src/autoscaler/main.py

```diff
@@ -0,0 +1,112 @@
+#!/usr/bin/python3
+"""
+Define the autoscaling service of the Scheduler.
+
+Classes:
+    AutoScalerService
+"""
+
+import math
+import subprocess
+from providers.main import Provider
+from scheduler.api import APIService
+from providers.replica import ReplicaStatus
+from autoscaler.settings import AUTOSCALING_MIN_REPLICA
+
+
+class AutoScalerService:
+    """Scheduler autoscaling service of the scaling module."""
+
+    provider: Provider
+    api: APIService
+    replica_capacity: int
+    min_available_resources: int
+    master: bool
+    address: str
+
+
+    def __init__(self, provider: Provider, port: int):
+        self.provider = provider
+        self.api = APIService({
+            "protocol": "http",
+            "port": port,
+            "path": "/",
+            "capacity_key": "status",
+            "termination_key": "isMaster"
+        })
+        self.replica_capacity = 1
+        self.min_available_resources = AUTOSCALING_MIN_REPLICA
+        self.master = False
+        self.address = subprocess.getoutput('hostname -I | grep -o -E "([0-9]{1,3}\.){3}[0-9]{1,3}" 2>&1 | head -n 1')
+
+    def getMaster(self):
```

Comment:
```suggestion
    def get_master(self):
```
And change all other occurrences